okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -396,20 +393,19 @@ private void sendBuffer(Buffer buffer, boolean endOfStream, boolean flush) {
       }
     }
 
-    @SuppressWarnings("GuardedBy")
     @GuardedBy("lock")
     private void streamReady(Metadata metadata, String path) {
-      requestHeaders =
-          Headers.createRequestHeaders(
-              metadata,
-              path,
-              authority,
-              userAgent,
-              useGet,
-              transport.isUsingPlaintext());
-      // TODO(b/145386688): This access should be guarded by 'this.transport.lock'; instead found:
-      // 'this.lock'
-      transport.streamReadyToStart(OkHttpClientStream.this);
+      synchronized (OkHttpClientTransport.lock) {
```

Comment:
why can't we add synchronized keyword in the method itself if everything in the method needs to be in synchronized block?